fv3core/utils/grid.py

```diff
@@ -412,3 +602,46 @@ def axis_offsets(
         "j_end": j_end,
         "local_je": gtscript.J[-1] + grid.je - origin[1] - domain[1] + 1,
     }
+
+
+@functools.lru_cache(maxsize=None)
+def _grid_indexing_axis_offsets(
+    grid: GridIndexing,
+    origin: Tuple[int, ...],
+    domain: Tuple[int, ...],
+) -> Mapping[str, gtscript._AxisOffset]:
```

Comment:
@jdahm This should be a public class in gtscript, because it's the type of a publicly available variable.

Reply:
Yes, it will be called `gtscript.AxisIndex` soon-ish.